### PR TITLE
Update feedstock for 2.x dependencies

### PR DESCRIPTION
**Pull request by Sahiler**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [X] Reset the build number to `0` (if the version changed)
* [X] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by jacksund** [comment]
@Sahiler thanks for working on this! fyi make sure you're checking the errors from failed CI. They are indicating that you have [mismatched SHA256 keys](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=544601&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=e5c8ab1d-8ff9-5cae-b332-e15ae582ed2d&l=433)

**Comment by Sahiler** [comment]
Do you know what the linter error is referring to/ can you provide more information?

**Comment by jacksund** [comment]
You can remove the python version restriction to get rid of the linting error (so change the line `python >=3.7` to just `python`). Everything before 3.7 has hit end of life already (see [here](https://endoflife.date/python)), so this should be fine to drop. 

If the prefect team doesn't like this, you can also make the feedstock more robust by adding a 'pip check' to your tests. You can use this section of one of my feedstocks as an example:
https://github.com/conda-forge/simmate-feedstock/blob/main/recipe/meta.yaml#L58-L62

**Comment by jacksund** [comment]
instead of `simmate --help` in my example, you can use `prefect --help`

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,58 +1,73 @@
...
   version: {{ version }}
 
 source:
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: 774a50684cd8ef365d071ed1a4f2249213559fed40e7a0fd9ef36a51009e3064
+  sha256: b068d6446ab7c227f703e4670862274ce3eea14e2925a41eae48c184fdf32ce2
 
 build:
-  noarch: python
+  skip: True  # [py==38 and py==310]
```
Comment (madkinsz) [inline recipe/meta.yaml:13]: Can you explain this one?

Comment (Sahiler) [reply recipe/meta.yaml]: It was used to skip builds with python versions of 3.8 and 3.10 in order to not download the affiliated packages per the Orion [requirements.txt](https://github.com/PrefectHQ/orion/blob/main/requirements.txt) file

**Comment by madkinsz** [comment]
We can add pip check / help tests separately, it's nice to keep this scoped to updating the requirements.

@Sahiler is there a reason you removed `noarch: python`? I believe that was providing an architecture agnostic/generic package build.

**Pull request merged**